fpsim/wrappers.py

```diff
@@ -0,0 +1,1216 @@
+        **What this models:** Changes in how well a method prevents pregnancy, which could result from:
+        - Improved product quality (e.g., better manufacturing standards)
+        - Better user education (e.g., teaching correct condom use)
+        - Improved formulations (e.g., higher-dose pills)
+        
+        Parameters
+        ----------
+        method : str
+            Method name: 'pill', 'iud', 'inj' (injectables), 'impl' (implants), 
+            'cond' (condoms), 'btl' (tubal ligation), 'wdraw' (withdrawal), etc.
+        efficacy : float
+            Effectiveness as a decimal between 0 and 1
+            Examples: 0.95 = 95% effective, 0.99 = 99% effective
+            
+        Returns
+        -------
+        self : MethodIntervention
+            Returns self for method chaining (so you can do multiple .set_X() calls)
+            
+        Examples
+        --------
+        >>> # Model improved pill quality
+        >>> mod = MethodIntervention(year=2025)
+        >>> mod.set_efficacy('pill', 0.95)  # Increase from baseline to 95%
+        
+        >>> # Model better condom education
+        >>> mod.set_efficacy('cond', 0.96)  # Increase correct use effectiveness
+        
+        Notes 
+        --------------------
+        - Typical efficacies: Pills ~94%, IUDs ~99%, Implants ~99%, Condoms ~95%
+        - Small changes (2-3%) can have meaningful impact on unintended pregnancies
+        - Efficacy changes affect pregnancy outcomes but don't directly change method mix
+        """
+        # endregion: documentation
+        
+        val = float(efficacy)
+        if not (0.0 <= val <= 1.0):
+            raise ValueError(f'Efficacy for {method} must be between 0 and 1')
+        
+        with _print_if(print_efficacy, f'Efficacy: {method} = {val}'):
+            self._config.eff[method] = val
+        
+        return self
+
+    @validate_method_name(allow_new=True)
+    def set_duration_months(self, method: str, months: Union[int, float], print_duration=False) -> 'MethodIntervention':
+        # region: docstring
+        """
+        Set how long (in months) people typically stay on a contraceptive method.
+        
+        **What this models:** Continuation rates - how long people keep using a method before 
+        stopping. This is one of the MOST IMPACTFUL interventions you can model. Improvements 
+        could result from:
+        - Better counseling and side effect management
+        - Improved supply chain (reducing stockouts)
+        - Reduced side effects (better formulations)
+        - Better follow-up and support services
+        - Community education reducing stigma
+        
+        Parameters
+        ----------
+        method : str
+            Method name: 'pill', 'iud', 'inj', 'impl', 'cond', 'btl', 'wdraw', etc.
+        months : int or float
+            Average duration in months that people stay on the method
+            Examples: 12 = 1 year, 24 = 2 years, 36 = 3 years
```

Comment:
Ideally these should all use Starsim's time logic, so it'd be something like ss.months(30) and then it'd get translated to the right number of timesteps depending on what `dt` is.